services/prototype.go

```diff
@@ -289,8 +289,29 @@ type SearchDatabaseInput struct {
 	SearchDatabaseInputWithoutHeader
 }
 
+// routes database-related errors through Huma
+func databaseError(err error) error {
+	if err != nil {
+		slog.Error(err.Error())
+		switch err.(type) {
+		case *databases.InvalidSearchParameter:
+			return huma.Error400BadRequest(err.Error(), err)
+		case *databases.UnavailableError:
+			return huma.Error503ServiceUnavailable(err.Error(), err)
+		case *databases.PermissionDeniedError, *databases.UnauthorizedError:
+			return huma.Error401Unauthorized(err.Error(), err)
+		case *databases.NotFoundError, *databases.ResourceNotFoundError, *databases.ResourceEndpointNotFoundError:
+			return huma.Error404NotFound(err.Error(), err)
+		default:
+			return huma.Error500InternalServerError(err.Error(), err)
+		}
+	} else {
+		return nil
+	}
+}
+
 // implements database search for both GET and POST requests
-func searchDatabase(ctx context.Context,
+func searchDatabase(_ context.Context,
```

Comment:
is the context object not used, then?

Reply:
That's correct. But I can revisit and see whether this arg is actually needed.

Reply:
If it's not needed, might be simplest to remove it so future devs aren't confused as to why there's an unused arg.

Reply:
Agreed. I can evaluate and address this in a cleanup PR (coming soon!).